public/app/plugins/panel/trend/TrendPanel.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useMemo } from 'react';
+
+import { PanelProps } from '@grafana/data';
+import { config, PanelDataErrorView } from '@grafana/runtime';
+import {
+  KeyboardPlugin,
+  TimeSeries,
+  TooltipDisplayMode,
+  TooltipPlugin,
+  usePanelContext,
+  ZoomPlugin,
+} from '@grafana/ui';
+
+import { ContextMenuPlugin } from '../timeseries/plugins/ContextMenuPlugin';
+import { prepareGraphableFields, regenerateLinksSupplier } from '../timeseries/utils';
+
+import { PanelOptions } from './panelcfg.gen';
+
+export const TrendPanel = ({
+  data,
+  timeRange,
+  timeZone,
+  width,
+  height,
+  options,
+  fieldConfig,
+  onChangeTimeRange,
+  replaceVariables,
+  id,
+}: PanelProps<PanelOptions>) => {
+  const { sync } = usePanelContext();
+
+  const frames = useMemo(() => prepareGraphableFields(data.series, config.theme2, timeRange), [data, timeRange]);
+
+  if (!frames) {
+    return (
+      <PanelDataErrorView
+        panelId={id}
+        fieldConfig={fieldConfig}
+        data={data}
+        needsTimeField={true}
+        needsNumberField={true}
+      />
+    );
+  }
+
+  return (
+    <TimeSeries // Name change!
```

Comment:
We can/should likely rename this component --it is shared between Trend+TimeSeries

Reply:
timeseries data _is_ a trend. the internal component can be renamed to Trend and it will have either x=time or x=number